daemon/CryptoImpl/crypto.cpp

```diff
@@ -1750,6 +1801,25 @@ void Daemon::ApiImpl::CryptoRequestQueue::handleFinishedRequest(
             }
             break;
         }
+        case QueryLockStatusRequest: {
+            Result result = request->outParams.size()
+                    ? request->outParams.takeFirst().value<Result>()
+                    : Result(Result::UnknownError,
+                             QLatin1String("Unable to determine result of QueryLockStatusRequest request"));
+            if (result.code() == Result::Pending) {
+                // shouldn't happen!
+                qCWarning(lcSailfishCryptoDaemon) << "QueryLockStatusRequest:" << request->requestId << "finished as pending!";
+                *completed = true;
+            } else {
+                bool locked = request->outParams.size()
+                        ? request->outParams.takeFirst().value<bool>()
+                        : true;
+                request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
+                                                                        << QVariant::fromValue<bool>(locked));
+                *completed = true;
```

Comment:
Both branches have *completed = true could be set before break.

Reply:
will fix